src/server/controllers/votes.js

```diff
@@ -1,17 +1,93 @@
 import { respondWithSuccess } from '~/server/helpers/respond';
 import dispatch from '~/server/services/dispatcher';
-import requestGraph from '~/common/services/subgraph';
 import Vote from '~/server/models/vote';
-import Answer from '~/server/models/answer';
+import { QuestionHasManyAnswers } from '~/server/database/associations';
+import Question from '~/server/models/question';
 import baseController from '~/server/controllers';
+import {
+  filterResponseFieldsAll,
+  filterResponseFields,
+} from '~/server/controllers';
+
+const customFilter = (req, data, options) => {
+  let combined = combineWithGraph(
+    req.locals.graphData,
+    data.answers,
+    'chainId',
+  );
+  if (req.locals && req.locals.user) {
+    data.set('answers', combined, {
+      raw: true,
+    });
+    return filterResponseFieldsAll(req, data, options);
+  }
+  const topThree = findTopThree(combined, 'votePower');
+  combined = combined.map(datum => {
+    if (topThree.includes(parseInt(datum['votePower']))) {
+      const optionsOverride = {
+        ...options,
+        fields: options.fields.concat(options.fieldsProtected),
+      };
+      return filterResponseFields(req, datum, optionsOverride);
+    }
+    return filterResponseFields(req, datum, options);
+  });
+  data.set('answers', combined, {
+    raw: true,
+  });
+  return data;
+};
 
 const options = {
-  model: Answer,
-  fields: ['id', 'votes', 'voteTokens', 'votePower', 'question'],
-  thresholdFields: ['type', 'artworkId', 'propertyId'],
-  threshold: 3,
-  thresholdKey: 'votePower',
-  graphMatchingKey: 'chainId',
+  model: Question,
+  fields: [
+    'title',
+    'address',
+    'answers',
+    'votes',
+    'voteTokens',
+    'votePower',
+    'question',
+  ],
+  fieldsProtected: ['id', 'chainId', 'type', 'artworkId', 'propertyId'],
+  include: [QuestionHasManyAnswers],
+  associations: [
+    {
+      association: QuestionHasManyAnswers,
+      destroyCascade: true,
+    },
+  ],
+  customFilter,
+};
+
+const combineWithGraph = (graphData, apiData, matchingKey) => {
+  const combined = graphData.map(i => {
+    const datum = apiData.find(d => d[matchingKey] === i.id);
+    if (datum) return Object.assign(datum.dataValues, i);
```

Comment:
Would prefer to not use inline if statements as they are hard to distinct from the rest of the code and invitive for bugs (https://dwheeler.com/essays/apple-goto-fail.html :smile_cat:)

Reply:
And lets not use short variable names like `i`, in this case I actually thought first its a number :-D